sklearn/compose/_column_transformer.py

```diff
@@ -212,6 +214,19 @@ class ColumnTransformer(TransformerMixin, _BaseComposition):
 
     _required_parameters = ["transformers"]
 
+    _parameter_constraints: dict = {
+        "transformers": [list],
+        "remainder": [
+            StrOptions({"drop", "passthrough"}),
+            HasMethods(["fit", "transform"]),
```

Comment:
technically it's a bit different than what was assumed before. This imposes to have ``fit`` and ``transform`` while it was allowed to only have ``fit_transform`` and ```transform``. There's a small risk that it will break users code if used on third party estimators that don't have ``fit``